router/router.go

```diff
@@ -2080,11 +2123,19 @@ func (*HandleT) crashRecover() {
 
 func Init() {
 	loadConfig()
+	setQueryFilters()
 	pkgLogger = logger.NewLogger().Child("router")
-	QueryFilters = jobsdb.QueryFiltersT{CustomVal: true}
 	Diagnostics = diagnostics.Diagnostics
 }
 
+func setQueryFilters() {
+	if readPerDestination {
+		QueryFilters = jobsdb.QueryFiltersT{CustomVal: true, ParameterFilters: []string{"destination_id"}}
```

Comment:
After having a quick look to our cache implementation in jobsDB, my impression is that we are going to introduce the following problem:

- Router jobsDB's cache will be operating with parameter filters in its maps, i.e. when it creates cache keys it will include the destination id in them.
- For routers where destination isolation is on, we will be querying jobsDB using destination id as a parameter filter, so cache will work as expected, we will have cache hits.
- For all other routers where destination isolation is off, we will be querying jobsDB without using destination id as a parameter filter, thus the cache will never find a hit, i.e. it will be as if we won't be using any cache at all.

Based on the above, I believe that we cannot be selective on destination isolation, either we need to enable it for all destination types or for none.

cc @Sidddddarth to further comment on the cache's behaviour if necessary

Reply:
actually, it might be even more complex than the description above, nevertheless the point is that it is unsafe to use a cache using a different filter configuration than the actual queries...